ADMVEDU95/ADMVEDU95/Presentation/Detail/Presenter/DetailPresenter.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  DetailPresenter.swift
+//  ADMVEDU95
+//
+//  Created by Satishur, Oleg on 27.04.2021.
+//
+
+import Foundation
+import AVFoundation
+import AVKit
+
+protocol DetailViewProtocol: class {
+    func setButtonImage(isPlayed: Bool)
+}
+
+protocol DetailViewPresenterProtocol: class {
+    init(view: DetailViewProtocol, dataKind: ResponseDataKind)
+    var dataKind: ResponseDataKind { get set }
+    var playerViewController: AVPlayerViewController { get set }
+    var player: AVPlayer? { get set }
+    var playerItem: AVPlayerItem? { get set }
+    func playVideo()
+    func playMusic()
+    func initAudioPlayer(songUrl: String?)
+    func initVideoPlayer(movieUrl: String?)
+}
+
+class DetailPresenter: DetailViewPresenterProtocol {
```

Comment:
Let's split this presenter (remember single responsibility principle)
It shouldn't mix song and movie